protocols/mdns/src/behaviour/iface.rs

```diff
@@ -171,17 +175,14 @@ impl InterfaceState {
 
                     let new_expiration = Instant::now() + peer.ttl();
 
-                    let mut addrs: Vec<Multiaddr> = Vec::new();
                     for addr in peer.addresses() {
-                        if let Some(new_addr) = address_translation(addr, &observed) {
-                            addrs.push(new_addr.clone())
-                        }
-                        addrs.push(addr.clone())
-                    }
+                        let da = if let Some(new_addr) = address_translation(addr, &observed) {
+                            new_addr.clone()
+                        } else {
+                            addr.clone()
+                        };
 
-                    for addr in addrs {
-                        self.discovered
-                            .push_back((*peer.id(), addr, new_expiration));
+                        self.discovered.push_back((*peer.id(), da, new_expiration));
```

Comment:
This is not functionally equivalent I think. Previously, we've pushed the translated AND untranslated address whereas now we only push one or the other, depending on whether the translation worked. What is the reason for the behaviour change?

Reply:
It is fine to reduce the two loops to one but we need to keep the same functionality :)

The code snipped you posted seems good!